### PR TITLE
[JENKINS-57775][JENKINS-62780] Handle PRs from/to non existing src/dest

### PR DESCRIPTION
**Pull request by Dohbedoh**
Fix NPEs caused by deleted source and destination branch or repository of PRs.

* [JENKINS-62780](https://issues.jenkins-ci.org/browse/JENKINS-62780)
* [JENKINS-57775](https://issues.jenkins-ci.org/browse/JENKINS-57775)
* https://github.com/jenkinsci/bitbucket-branch-source-plugin/pull/283

Tested the different scenario against Bitbucket Cloud and confirmed that:

* `source.repository` can be `null` (deleted source repository)
* `source.commit` can be `null` (deleted source repository)
* `destination.commit` can be `null` (deleted destination branch)

I haven't been able to reproduce a nullable `source.branch` or `destination.branch`.

When testing against Bitbucket Server (7.2.1), I could not reproduce such scenario:

* PR opened from deleted source branch or forked are automatically declined.
* Bitbucket Server does not let me delete a destination branch if there is an open PR against it: "You cannot delete this branch since it has at least one open pull request associated with it"

### Your checklist for this pull request

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side) and not your master branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or in [Jenkins JIRA](https://issues.jenkins-ci.org)
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Did you provide a test-case? That demonstrates feature works or fixes the issue.

### CODE REVIEW

File: src/main/java/com/cloudbees/jenkins/plugins/bitbucket/client/pullrequest/BitbucketPullRequestValueDestination.java [hunk 1]
```diff
@@ -44,14 +45,14 @@
...
+                                                @JsonProperty("commit") BitbucketCloudCommit commit) {
         this.repository = repository;
         this.branch = branch;
         this.commit = commit;
 
         // It is possible for a PR's original destination to no longer exist.
-        if(this.branch != null) {
-            // redound available the informations into impl objects
+        if(this.branch != null && this.commit != null) {
+            // redound available the information into impl objects
```
Comment (bitwiseman) [inline src/main/java/com/cloudbees/jenkins/plugins/bitbucket/client/pullrequest/BitbucketPullRequestValueDestination.java:55]: "redound"?

File: src/main/java/com/cloudbees/jenkins/plugins/bitbucket/client/pullrequest/BitbucketPullRequestValueDestination.java [hunk 2]
```diff
@@ -44,14 +45,14 @@
     @JsonCreator
     public BitbucketPullRequestValueDestination(@NonNull @JsonProperty("repository") BitbucketCloudRepository repository,
                                                 @JsonProperty("branch") BitbucketCloudBranch branch,
-                                                @NonNull @JsonProperty("commit") BitbucketCloudCommit commit) {
+                                                @JsonProperty("commit") BitbucketCloudCommit commit) {
```
Comment (bitwiseman) [inline src/main/java/com/cloudbees/jenkins/plugins/bitbucket/client/pullrequest/BitbucketPullRequestValueDestination.java:48]:
```suggestion
                                                @CheckForNull @JsonProperty("commit") BitbucketCloudCommit commit) {
```

File: src/main/java/com/cloudbees/jenkins/plugins/bitbucket/client/pullrequest/BitbucketPullRequestValueDestination.java (hunk 1 above)
Comment (Dohbedoh) [reply src/main/java/com/cloudbees/jenkins/plugins/bitbucket/client/pullrequest/BitbucketPullRequestValueDestination.java]: That's from https://github.com/jenkinsci/bitbucket-branch-source-plugin/pull/152. I am gonna rephrase this to `Make the commit information available into impl objects`. cc @nfalco79

Comment (nfalco79) [reply src/main/java/com/cloudbees/jenkins/plugins/bitbucket/client/pullrequest/BitbucketPullRequestValueDestination.java]: It should was " re bind"

Comment (nfalco79) [reply src/main/java/com/cloudbees/jenkins/plugins/bitbucket/client/pullrequest/BitbucketPullRequestValueDestination.java]: @bitwiseman I want heavy remember to you [branch-api-plugin#160](https://github.com/jenkinsci/branch-api-plugin/pull/160)

**Comment by Dohbedoh** [comment]
> Since you're controlling/mocking the responses, you could create a custom one that has these behaviors right?

I am not sure that those behaviors are even possible from Bitbucket Server.

File: src/main/java/com/cloudbees/jenkins/plugins/bitbucket/client/pullrequest/BitbucketPullRequestValueDestination.java (hunk 1 above)
Comment (bitwiseman) [reply src/main/java/com/cloudbees/jenkins/plugins/bitbucket/client/pullrequest/BitbucketPullRequestValueDestination.java]: @nfalco79  I hear you.
